manifests/keypair.pp

```diff
@@ -0,0 +1,41 @@
+define certs::keypair (
+  $key_pair,
+  $key_file,
+  $cert_file,
+  $manage_key  = false,
+  $key_owner   = undef,
+  $key_group   = undef,
+  $key_mode    = undef,
+  $manage_cert = false,
+  $cert_owner  = undef,
+  $cert_group  = undef,
+  $cert_mode   = undef,
+) {
+  Cert[$key_pair] ~>
```

Comment:
I do believe this is redundant but didn't want to drop too much at the same time. This at least puts it in one place and makes it easier to change it.